extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -0,0 +1,187 @@
+package io.druid.emitter.graphite;
+
+import com.codahale.metrics.graphite.PickledGraphite;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.metamx.common.ISE;
+import com.metamx.common.logger.Logger;
+import com.metamx.emitter.core.Emitter;
+import com.metamx.emitter.core.Event;
+import com.metamx.emitter.service.AlertEvent;
+import com.metamx.emitter.service.ServiceMetricEvent;
+
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+public class GraphiteEmitter implements Emitter
+{
+  private static Logger log = new Logger(GraphiteEmitter.class);
+
+  private final GraphiteEventConverter graphiteEventConverter;
+  private final GraphiteEmitterConfig graphiteEmitterConfig;
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ObjectMapper mapper;
+  private final LinkedBlockingQueue<GraphiteEvent> eventsQueue;
+  private final static long DEFAULT_PUT_GET_TIMEOUT = 1000; // default wait for put/get operations on the queue 1 sec
+  private static final long FLUSH_TIMEOUT = 60000; // default flush wait 1 min
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(
+      2,
+      new ThreadFactoryBuilder().setDaemon(
+          true)
+                                .setNameFormat(
+                                    "GraphiteEmitter-%s")
+                                .build()
+  ); // Thread pool of two in order to schedule flush runnable
+
+  public GraphiteEmitter(@NotNull GraphiteEmitterConfig graphiteEmitterConfig, @NotNull ObjectMapper mapper)
+  {
+    this.mapper = mapper;
+    this.graphiteEmitterConfig = graphiteEmitterConfig;
+    this.graphiteEventConverter = new GraphiteEventConverter(
```

Comment:
Since we are using dependency injection to inject dependencies in the constructor,should we inject GraphiteEventConverter as well??